modules/virt-measuring-latency-vm-secondary-network.adoc

```diff
@@ -0,0 +1,229 @@
+* You logged in to the cluster as a user with the `cluster-admin` role.
+* The cluster has at least two worker nodes.
+* The Multus Container Network Interface (CNI) plug-in is installed on the cluster.
+* You configured a Linux bridge network attachment definition for a namespace.
+
+.Procedure
+
+. Create a configuration file that contains the resources to set up the framework. This includes a namespace and service account for the framework, and the `ClusterRole` and `ClusterRoleBinding` objects to define permissions for the service account.
++
+.Example framework manifest file
+[%collapsible]
+====
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: k8s-checkup-framework
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: checkup-framework-sa
+  namespace: k8s-checkup-framework
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: cluster-admin-to-checkup-framework-sa
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+- kind: ServiceAccount
+  name: checkup-framework-sa
+  namespace: k8s-checkup-framework
+----
+====
+
+. Apply the framework manifest:
++
+[source,terminal]
+----
+$ oc apply -f <framework-manifest>.yaml
+----
+
+. Create a configuration file that contains the `ClusterRole` and `Role` objects with permissions that are required by the checkup to access the cluster:
++
+.Example checkup role manifest file
+[%collapsible]
+====
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: nodes-reader
+rules:
+- apiGroups: [""]
+  resources: ["nodes"]
+  verbs: ["get"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kubevirts-reader
+rules:
+- apiGroups: ["kubevirt.io"]
+  resources:
+  - kubevirts
+  verbs:
+  - get
+  - list
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: networks-reader
+rules:
+- apiGroups: ["k8s.cni.cncf.io"]
+  resources: ["network-attachment-definitions"]
+  verbs: ["get"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kubevirt-vmis-manager
+rules:
+- apiGroups: ["kubevirt.io"]
+  resources: ["virtualmachineinstances"]
+  verbs:
+  - get
+  - create
+  - delete
+- apiGroups: ["subresources.kubevirt.io"]
+  resources: ["virtualmachineinstances/console"]
+  verbs: ["get"]
+----
+====
+
+. Apply the checkup roles manifest:
++
+[source,terminal]
+----
+$ oc apply -f <latency-roles>.yaml
+----
+
+. Create a `ConfigMap` manifest that contains the environment variables for the checkup. The config map provides input for the framework to run the checkup and also stores the results of the checkup.
++
+.Example input config map
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kubevirt-latency-checkup-example-config
+  namespace: k8s-checkup-framework
+data:
+  spec.image: registry:5000/kubevirt-latency-checkup:latest
```

Comment:
@orelmisan , this is the correct ref to the D/S image?

Reply:
The M/S image name is `vm-network-latency-checkup`, I'll check what the full D/S name is.

Reply:
`registry.redhat.io/container-native-virtualization/vm-network-latency-checkup:v4.11.0`